src/profile/useUserProfile.test.ts

```diff
@@ -162,7 +162,7 @@ describe('useUserProfile', () => {
       asMockedFn(User).mockImplementation(() => {
         return {
           profile: {
-            set: setProfile,
+            update: setProfile,
```

Comment:
Nit: suggest renaming the mock to match.

```suggestion
            update: updateProfile,
```